src/vscode/commands/Repl.ts

```diff
@@ -232,6 +278,152 @@ export async function loadFile(state: ExtensionState) {
+    const response = await axios(latest.zipball_url, {
+        headers: { 'User-Agent': 'nobody-famous/alive' },
+        method: 'GET',
+        responseType: 'stream',
+    })
+
+    await fs.promises.mkdir(zipPath, { recursive: true })
+    await new Promise((resolve, reject) => {
+        const writer = fs.createWriteStream(zipFile)
+        response.data.pipe(writer).on('finish', resolve).on('close', resolve).on('error', reject)
+    })
+    const streamZip = new StreamZip.async({ file: zipFile })
+    await streamZip.extract(null, zipPath)
+    await streamZip.close()
+
+    const slimePath = await getSlimePath(state, latest.name)
+    if (!slimePath) {
+        throw new Error('Failed to download latest Slime version')
+    }
+
+    return { path: slimePath, latest }
+}
+
+async function getInstalledVersionName(state: ExtensionState): Promise<string | undefined> {
+    const files = await fs.promises.readdir(getSlimeBasePath(state))
+    if (files.length !== 1) {
+        await fs.promises.rm(getSlimeBasePath(state), { recursive: true })
+        return undefined
+    }
+    return files[0]
+}
+
+async function getSlimePath(state: ExtensionState, versionName: string): Promise<string | undefined> {
+    const files = await fs.promises.readdir(path.join(getSlimeBasePath(state), versionName))
+    if (files.length !== 2) {
+        await fs.promises.rm(getSlimeBasePath(state), { recursive: true })
+        return undefined
+    }
+    const zipFileName = `${path.basename(versionName)}.zip`
+    const hashDirectory = files.filter((f) => f !== zipFileName)[0]
+    return path.join(getSlimeBasePath(state), versionName, hashDirectory)
+}
+
+function getClSourceRegistryEnv(
+    slimePath: string,
+    processEnv: { [key: string]: string | undefined }
+): { [key: string]: string | undefined } {
+    const updatedEnv = { ...processEnv }
+    if (!processEnv.CL_SOURCE_REGISTRY) {
+        updatedEnv.CL_SOURCE_REGISTRY = slimePath
+        return updatedEnv
+    }
+    if (processEnv.CL_SOURCE_REGISTRY.startsWith('(')) {
+        const slimePathSExpressionEnding = ` (:directory "${slimePath}")`
+        updatedEnv.CL_SOURCE_REGISTRY = `${processEnv.CL_SOURCE_REGISTRY.replace(/\)$/, slimePathSExpressionEnding)})`
+        return updatedEnv
+    }
+    updatedEnv.CL_SOURCE_REGISTRY = `${processEnv.CL_SOURCE_REGISTRY}${path.delimiter}${slimePath}`
+    return updatedEnv
+}
+
+async function getWorkspaceOrFilePath(): Promise<string> {
+    if (vscode.workspace.workspaceFolders === undefined) {
+        return path.dirname(vscode.window.activeTextEditor?.document.fileName || homedir())
+    }
+
+    const folder =
+        vscode.workspace.workspaceFolders.length > 1
+            ? await pickWorkspaceFolder(vscode.workspace.workspaceFolders)
+            : vscode.workspace.workspaceFolders[0]
+
+    if (folder === undefined) {
+        throw new Error('Failed to find a workspace folder')
+    }
+
+    return folder.uri.path
```

Comment:
This needs to be

return folder.uri.fsPath

On my Windows machine, it causes spawn to fail because the path is malformed. Apparently fsPath is a getter that formats it correctly.

Reply:
Ugh, FML, I didn't have watch running when I tested it on Windows so I never validated any of the changes I made after I made it handle things getting torn out from under it.  I'll fix this now.  Sorry to waste your time.